content/technology/jest/_index.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Jest
+description: How to write jest test in altinn studio, what we know so far
+weight: 100
+tags: ["tjenester 3.0"]
+---
+
+Here you can find information related to have jest tests are written for altinn studio code.
+
+#### Exporting your component
+To be able to test methods in your component the best way is to export it in the following way:
+``` javascript
+export class CloneServiceComponent extends React.Component<ICloneServiceComponentProps & RouteChildrenProps, ICloneServiceComponentState> {
+```
+
+and reference it in your test the following way:
+``` javascript
+import { CloneServiceComponent } from '../../../src/dashboardServices/cloneService/cloneServices';
+
+  it('+++ this is the test', () => {
+
+    const mountedComponent = mount(
+      <CloneServiceComponent
+        language={mockLanguage}
+        services={mockServices}
+        classes={mockClasses}
+        location={mockLocation}
+        history={mockHistory}
+        match={mockMatch}
+      />,
+    );
+  
+    const instance = mountedComponent.instance() as CloneServiceComponent;
+  }
+```
+This will test the component without state, router or styles (read Material UI),
+so values usually passed in by redux store, by means of routing or styles/Material UI needs to be reference as if they are props passed into the component (match={mockMatch}).
+
+#### Testing your component
+Methods can be called directly from your test, so you don't necessarly have to simulate an event to test your code. This is done the following way:
+
+``` javascript
+ instance.getCurrentRepositoryInfo();
+ instance.componentDidMount();
+```
+
+In some cases it might be necessary to run componentDidMount before your acctual test since enzymes mount functions acctually does not do that for you. 
+
+If you have one function that calls an other function, and you want to be sure that both funciton ran at the end of the test, you can use jest's spyOn functionality the following way:
+
+``` javascript
+ const spy = jest.spyOn(instance, 'funcitonTwoCalledWithinFunctionOne');
+ instance.funcitonOne();
+ expect(spy).toHaveBeenCalled();
+```
+
+It might also be usefull to take alook at [jest expect](https://jestjs.io/docs/en/expect) for inspiration on other test methods to run expect on
```

Comment:
You can read more about Jest.expect() here: [Jest.expect()](https://jestjs.io/docs/en/expect)